aws-gitlab-main.yml

```diff
@@ -467,14 +502,49 @@ Resources:
     Type: AWS::EC2::Instance
     Properties: 
       AvailabilityZone: !Ref BastionHostBAvailabilityZoneParameter 
-      
       ImageId: !Ref GitLabBastionHostBAMIParameter 
       InstanceType: !Ref GitLabBastionHostBInstanceTypeParameter
       SubnetId: !Ref GitLabPublicSubnetB
+      SecurityGroupIds: 
+        - !Ref GitLabBastionHostSecurityGroup  
       Tags: 
         - Key: Name
           Value: Bastion Host B 
+
+  #SSH AGENT FORWARDING??
```

Comment:
there are certain things you can do on runtime.... look into User Data for the instance which are commands it will run during boot only.

Rest make README.md and document steps taht need to be performed after its deployed to get it ready at running state. Do those steps yourselves and see what happens...

Reply:
Does that mean making a script to do the customization still inside the cft? 
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html

I saw something about remotely running commands using SSM..? I still don't get what it's supposed to look like.

Are you saying the steps after setting up Gitaly (fast lookup for SSH keys, configure host keys, S3 object storage...) are supposed to be instructions for README?

Reply:
Its either or.....
I would say give the user-data a try... see if that works.
if not see if cloud-init.

Do note user data should work but cloud init and SSM I believe require an agent in the instance

Amazon linux 2 has this agent but ubuntu does not. You will need to install it via user data to configure it.

So steps are:
1. Do via user data and see if that works, you put the bash commands there and push that through. Ton of examples on the web.
2. if it does not work >> use user data to install SSM agent 
3. than use SSM agent to do the steps needed

If all fails(aka too hard for u atm):
Just document all the steps in detail via readme